lib/cdo/contact_rollups_validation.rb

```diff
@@ -0,0 +1,308 @@
+              LIKE '%Facilitator%'",
+      min: 400,
+      max: 4000
+    },
+    {
+      name: "Workshop Organizer count",
+      query: "SELECT COUNT(*) from contact_rollups_daily WHERE roles
+              LIKE '%Workshop Organizer%'",
+      min: 250,
+      max: 2500
+    },
+    {
+      name: "District Contact count",
+      query: "SELECT COUNT(*) from contact_rollups_daily WHERE roles
+              LIKE '%District Contact%'",
+      min: 250,
+      max: 2500
+    },
+    {
+      name: "Regional Partner count",
+      query: "SELECT COUNT(*) from contact_rollups_daily where roles
+              LIKE '%Regional Partner%'",
+      min: 50,
+      max: 500
+    },
+    {
+      name: "Petition Signer count",
+      query: "SELECT COUNT(*) from contact_rollups_daily WHERE roles
+              LIKE '%Petition Signer%'",
+      min: 1_750_000,
+      max: 2_250_000
+    },
+    {
+      name: "Form Submitter count",
+      query: "SELECT COUNT(*) from contact_rollups_daily WHERE roles
+              LIKE '%Form Submitter%'",
+      min: 2_500_000,
+      max: 3_250_000
+    },
+    {
+      # expected # of distinct state values is 50 states + DC + null = 52
+      name: "Distinct US state count",
+      query: "SELECT COUNT(distinct state) FROM contact_rollups_daily
+              WHERE country = 'united states'",
+      min: 52,
+      max: 52
+    },
+    {
+      # Check that rollup's 'opt_out' data matches pegasus.contact's
+      # 'unsubscribed_at' data. In general the
+      # discrepancy should be 0 records. Allow for a handful of mismatches
+      # since production data may have changed slightly.
+      name: "Count of email addresses in rollup not marked as 'opted_out'"\
+            " that have 'unsubscribed_at' set in contacts",
+      query: "SELECT COUNT(*) FROM contact_rollups_daily
+              INNER JOIN contacts on contacts.email =
+              contact_rollups_daily.email
+              WHERE contacts.unsubscribed_at IS NOT NULL
+              AND contact_rollups_daily.opted_out <> 1",
+      min: 0,
+      max: 100
+    },
+    {
+      # max expected # of distinct state values is 50 states + null +
+      # 3 versions of "DC" =  54
+      name: "US state count",
+      query: "SELECT COUNT(distinct state) FROM contact_rollups_daily
+              WHERE country = 'united states'",
+      min: 50,
+      max: 54
+    },
+    {
+      name: "Country count",
+      query: "SELECT COUNT(distinct country) FROM contact_rollups_daily",
+      min: 200,
+      max: 250
+    },
+    {
+      name: "Count of contacts with non-empty names",
+      query: "SELECT COUNT(*) FROM contact_rollups_daily
+              WHERE name IS NOT NULL",
+      min: 3_000_000,
+      max: 10_000_000
+    },
+    {
+      name: "Count of contacts with NULL or empty names",
+      query: "SELECT COUNT(*) FROM contact_rollups_daily
+              WHERE name IS NULL OR LENGTH(name)=0",
+      min: 0,
+      max: 100000
```

Comment:
That was to prevent Rubocop from flagging dates (e.g., `2017` or `031317`). For any number that is a magnitude, I'd prefer to see `_` separators for readability.